### ISSUE TITLE
regularization

### ISSUE THREAD

**Issue by karaspd** [kujason/avod#94]
I can see l2_regularization parameters in layers, but I cannot see if it ever collected to add to loss function. Have you evaluated if regularization terms are used in the loss?

Thank you

**Comment by melfm** [comment]
The L2 regularization is defined as the sum of the squares of all the feature weights, please see the [equation](https://developers.google.com/machine-learning/crash-course/regularization-for-simplicity/l2-regularization). So it is passed as a parameter to `slim.arg_scope`.

**Comment by karaspd** [comment]
@melfm Thank you very much for replying. I was wondering if there is a way in tensorflow that you can ensure that regularization have been applied to gradients without bringing it in loss function explicitly.  From tensorflow.slim, I see they used slim.losses.get_total_loss() that will consider all losses include regularization, but in avod implementation all losses explicitly defined and summed. This makes me doubt that if avod ever used regularization term in the gradient.

**Comment by melfm** [comment]
I think we set it up and ended up not using it. You are more than welcome to try it and tune the parameter and report your findings.
